### PR TITLE
feat(sd): add process steps for sd creation

### PR DESCRIPTION
**Pull request by Phil91**
## Description

* add await response steps for the self description creation process
* add retrigger processes for the await response steps

## Why

To be able to display the current progress of the sd document recreation a await step has been added for the self description creation process

## Issue

Refs: #1182

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/admin/Dev%20Process/How%20to%20contribute.md#commits-branches-and-pull-requests-guidelines)
- [x] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [x] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/externalsystems/SdFactory.Library/BusinessLogic/SdFactoryBusinessLogic.cs [hunk 1]
```diff
@@ -152,21 +156,47 @@ public async Task ProcessFinishSelfDescriptionLpForConnector(SelfDescriptionResp
...
-            portalRepositories.GetInstance<IConnectorsRepository>().AttachAndModifyConnector(data.ExternalId, null, con =>
+            connectorsRepository.AttachAndModifyConnector(data.ExternalId, null, con =>
             {
                 con.SelfDescriptionMessage = data.Message!;
                 con.DateLastChanged = DateTimeOffset.UtcNow;
             });
         }
+
+        var processData = await connectorsRepository.GetProcessDataForConnectorId(data.ExternalId);
+        if (processData == null)
```
Comment (ntruchsess) [inline src/externalsystems/SdFactory.Library/BusinessLogic/SdFactoryBusinessLogic.cs:167]: this check processData == null guarantees that the subsequent call to CreateManualProcessData will throw NorFoundException. I suspect this is not intended ;-)

File: src/externalsystems/SdFactory.Library/BusinessLogic/SdFactoryBusinessLogic.cs [hunk 2]
```diff
@@ -152,21 +156,47 @@ public async Task ProcessFinishSelfDescriptionLpForConnector(SelfDescriptionResp
...
         if (data.Status == SelfDescriptionStatus.Confirm)
         {
             var documentId = await ProcessAndCreateDocument(SdFactoryResponseModelTitle.LegalPerson, data, cancellationToken).ConfigureAwait(ConfigureAwaitOptions.None);
-            portalRepositories.GetInstance<ICompanyRepository>().AttachAndModifyCompany(data.ExternalId, null,
-                c => { c.SelfDescriptionDocumentId = documentId; });
+            companyRepository.AttachAndModifyCompany(data.ExternalId, null, c => { c.SelfDescriptionDocumentId = documentId; });
+        }
+
+        var processData = await companyRepository.GetProcessDataForCompanyIdId(data.ExternalId);
+        if (processData == null)
```
Comment (ntruchsess) [inline src/externalsystems/SdFactory.Library/BusinessLogic/SdFactoryBusinessLogic.cs:197]: same here - checking processData == null seems to be incorrect.

**Review by ntruchsess — CHANGES_REQUESTED** [review]

**Review by ntruchsess — CHANGES_REQUESTED** [review]
In my previous review I mention 2 null checks that seem to be incorrect. I've fixed those but apparently this is not covered by unit-tests. Can you please double check and add respective test-code for those null-checks?

**Review by ntruchsess — APPROVED** [review]

**Review by ntruchsess — APPROVED** [review]
